pkg/transform/k8s.go

```diff
@@ -95,30 +98,89 @@ func KubeDecoratorProvider(
 	}
 }
 
+func ProcessEventDecoratorProvider(
```

Comment:
@mariomac Essentially I needed a decorator that decorates K8S or host based. It was easiest to place it here, but I'm thinking there might be a better place for this. Perhaps if we ever expand decorators to also fetch information from the known cloud vendors, we might need something like generic Metadata decorator, which decorates consistently depending on what's available.

What I'm thinking is if it makes sense to unify the read_decorator and the k8s decorator in a metadata decorator.

Reply:
If there aren't data dependencies or something that need to coordinated, maybe keeping two different nodes makes code simpler an easier to test/debug.